lib/std/crypto/benchmark.zig

```diff
@@ -21,6 +21,14 @@ const Crypto = struct {
     name: []const u8,
 };
 
+fn blackBox(x: anytype) void {
+    asm volatile (""
+        :
+        : [x] "rm" (x)
+        : "memory"
+    );
+}
```

Comment:
```suggestion
fn doNotOptimize(val: var) void {
    const T = @TypeOf(val);
    var store: T = undefined;
    @ptrCast(*volatile T, &store).* = val;
}
```
It's often used.
Can be useful as `pub fn` in `std.mem`?

Reply:
`s/: var/: anytype/`?

Reply:
Why not, but a copy to a `volatile` pointer adds at least an extra load. For arrays, we get a `memcpy` call that is really not required in order to prevent the compiler from doing unwanted optimizations.

That can indeed be useful to be in `std.mem`. Maybe we can come up with a more self-explanatory name? `doNotOptimize` looks like a function attribute rather than a way for a value to not be computed.

Maybe `consume`?

Reply:
I'm confused by `asm`.
Does it work for all targets?

Reply:
Yes, it works on all targets, including WebAssembly. No actual assembly code is emitted, but the constraints are respected.

Reply:
Re: naming, In [Chandler Carruth's talk on benchmarking](https://youtu.be/nXaxk27zwlk?t=2501) he calls this function `escape(x)`.  That's a pretty opaque name though.  [Google's Bench library](https://github.com/google/benchmark#preventing-optimization) calls this `DoNotOptimize(x)`.  Facebook's Folly library has [a slightly different implementation](https://github.com/facebook/folly/blob/master/folly/Benchmark.h#L315) and names it `doNotOptimizeAway(x)`.  `doNotOptimize` seems like a pretty standard name, I feel like it's pretty descriptive.